docs/resources/entitlements.md

```diff
@@ -0,0 +1,88 @@
+---
+subcategory: "Security"
+---
+# databricks_entitlements Resource
+
+This resource allows you to set entitlements to existing [databricks_users](user.md), [databricks_group](group.md) or [databricks_service_principal](service_principal.md)
+
+## Example Usage
+
+Setting entitlements for a regular user:
+
+```hcl
+data "databricks_user" "me" {
+  user_name = "me@example.com"
+}
+
+resource "databricks_entitlements" "me" {
+  user_id                    = data.databricks_user.me.id
+  allow_cluster_create       = true
+  allow_instance_pool_create = true
+}
+```
+
+Setting entitlements for a service principal:
+
+```hcl
+data "databricks_service_principal" "service_principal" {
+  application_id = "11111111-2222-3333-4444-555666777888"
+}
+
+resource "databricks_entitlements" "service_principal" {
+  service_principal_id       = data.databricks_service_principal.service_principal.sp_id
```

Comment:
```suggestion
resource "databricks_entitlements" "this" {
  service_principal_id       = data.databricks_service_principal.this.id
```

make names very short. `sp_id` field does not exist.

Reply:
sp_id field exists for the data source, as the id is _ for all data sources...